android/src/com/frostwire/android/gui/activities/ToggleAllSearchEnginesPreference2.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Created by Angel Leon (@gubatron), Alden Torres (aldenml)
+ * Copyright (c) 2011-2016, FrostWire(R). All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.frostwire.android.gui.activities;
+
+import android.content.Context;
+import android.graphics.Typeface;
+import android.preference.CheckBoxPreference;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.TextView;
+
+import com.frostwire.android.R;
+import com.frostwire.android.core.Constants;
+import com.frostwire.android.gui.SearchEngine;
+
+import java.util.Map;
+
+/**
+ * Created on 9/20/16.
+ *
+ * @author gubatron
+ * @author aldenml
+ * @author marcelinkaaa
+ * @author grzesiekrzaca
+ */
+public class ToggleAllSearchEnginesPreference2 extends CheckBoxPreference {
+
+    private CheckBox checkbox;
+    private Map<CheckBoxPreference, SearchEngine> searchEnginePreferences;
+    private int backgroundColor;
+    private boolean fixStateAfterNextBind = true;
+
+    private View.OnClickListener checkBoxListener = new View.OnClickListener() {
```

Comment:
why keep a reference to this `checkBoxListener` if it's not called anywhere else outside `onCreateView()`.

another way would be to have this class implement View.OnClickListener, and then the `onClick(View v)` implementation calls `onClick()`.

Let's try not to keep unnecessary properties and keep scopes as local as possible.